server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/DataTransferTask.java

```diff
@@ -0,0 +1,391 @@
+import com.oceanbase.odc.plugin.task.api.datatransfer.DataTransferCallable;
+import com.oceanbase.odc.plugin.task.api.datatransfer.dumper.DumpDBObject;
+import com.oceanbase.odc.plugin.task.api.datatransfer.dumper.DumperOutput;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferConfig;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferFormat;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferObject;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferTaskResult;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.DataTransferType;
+import com.oceanbase.odc.plugin.task.api.datatransfer.model.ObjectResult;
+import com.oceanbase.odc.service.common.util.SpringContextUtil;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.datasecurity.DataMaskingService;
+import com.oceanbase.odc.service.datasecurity.model.MaskingAlgorithm;
+import com.oceanbase.odc.service.datasecurity.model.SensitiveColumn;
+import com.oceanbase.odc.service.datasecurity.util.MaskingAlgorithmUtil;
+import com.oceanbase.odc.service.datatransfer.DataTransferAdapter;
+import com.oceanbase.odc.service.datatransfer.LocalFileManager;
+import com.oceanbase.odc.service.datatransfer.dumper.SchemaMergeOperator;
+import com.oceanbase.odc.service.datatransfer.model.DataTransferProperties;
+import com.oceanbase.odc.service.db.browser.DBSchemaAccessors;
+import com.oceanbase.odc.service.iam.model.User;
+import com.oceanbase.odc.service.iam.util.SecurityContextUtils;
+import com.oceanbase.odc.service.plugin.TaskPluginUtil;
+import com.oceanbase.odc.service.session.factory.DefaultConnectSessionFactory;
+import com.oceanbase.tools.dbbrowser.model.DBTableColumn;
+import com.oceanbase.tools.dbbrowser.schema.DBSchemaAccessor;
+import com.oceanbase.tools.loaddump.common.enums.ObjectType;
+import com.oceanbase.tools.loaddump.common.model.ObjectStatus.Status;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.extern.slf4j.Slf4j;
+
+@Builder
+@Slf4j
+public class DataTransferTask implements Callable<DataTransferTaskResult> {
+    private static final Set<String> OUTPUT_FILTER_FILES = new HashSet<>();
+
+    private final DataMaskingService maskingService;
+    private final Holder<DataTransferCallable> jobHolder;
+    private final User creator;
+    private final DataTransferConfig config;
+    private final DataTransferAdapter adapter;
+    private final DataTransferProperties properties;
+    private final File workingDir;
+    private final File logDir;
+    private final ConnectionConfig connectionConfig;
+
+    @Override
+    public DataTransferTaskResult call() throws Exception {
+        try {
+            SecurityContextUtils.setCurrentUser(creator);
+
+            List<URL> inputs = new ArrayList<>();
+            preHandle(inputs);
+
+            DataTransferCallable job = TaskPluginUtil
```

Comment:
this job may be a field of this object, using getter method to access it may be better

Reply:
done